crates/router/src/core/payments/operations/payment_complete_authorize.rs

```diff
@@ -0,0 +1,424 @@
+    async fn get_or_create_customer_details<'a>(
+        &'a self,
+        db: &dyn StorageInterface,
+        payment_data: &mut PaymentData<F>,
+        request: Option<CustomerDetails>,
+        merchant_id: &str,
+    ) -> CustomResult<
+        (
+            BoxedOperation<'a, F, api::PaymentsRequest>,
+            Option<storage::Customer>,
+        ),
+        errors::StorageError,
+    > {
+        helpers::create_customer_if_not_exist(
+            Box::new(self),
+            db,
+            payment_data,
+            request,
+            merchant_id,
+        )
+        .await
+    }
+
+    #[instrument(skip_all)]
+    async fn make_pm_data<'a>(
+        &'a self,
+        state: &'a AppState,
+        payment_data: &mut PaymentData<F>,
+        _storage_scheme: storage_enums::MerchantStorageScheme,
+    ) -> RouterResult<(
+        BoxedOperation<'a, F, api::PaymentsRequest>,
+        Option<api::PaymentMethodData>,
+    )> {
+        let (op, payment_method_data) =
+            helpers::make_pm_data(Box::new(self), state, payment_data).await?;
+
+        utils::when(payment_method_data.is_none(), || {
+            Err(errors::ApiErrorResponse::PaymentMethodNotFound)
+        })?;
+
+        Ok((op, payment_method_data))
+    }
+
+    #[instrument(skip_all)]
+    async fn add_task_to_process_tracker<'a>(
+        &'a self,
+        _state: &'a AppState,
+        _payment_attempt: &storage::PaymentAttempt,
+    ) -> CustomResult<(), errors::ApiErrorResponse> {
+        Ok(())
+    }
+
+    async fn get_connector<'a>(
+        &'a self,
+        _merchant_account: &storage::MerchantAccount,
+        state: &AppState,
+        request: &api::PaymentsRequest,
+        previously_used_connector: Option<&String>,
+    ) -> CustomResult<api::ConnectorCallType, errors::ApiErrorResponse> {
+        // Use a new connector in the confirm call or use the same one which was passed when
+        // creating the payment or if none is passed then use the routing algorithm
+        let request_connector = request
+            .connector
+            .as_ref()
+            .and_then(|connector| connector.first().map(|c| c.to_string()));
+        helpers::get_connector_default(
+            state,
+            request_connector.as_ref().or(previously_used_connector),
+        )
+        .await
+    }
+}
+
+#[async_trait]
+impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for CompleteAuthorize {
+    #[instrument(skip_all)]
+    async fn update_trackers<'b>(
+        &'b self,
+        db: &dyn StorageInterface,
+        _payment_id: &api::PaymentIdType,
+        mut payment_data: PaymentData<F>,
+        customer: Option<storage::Customer>,
+        storage_scheme: storage_enums::MerchantStorageScheme,
+    ) -> RouterResult<(BoxedOperation<'b, F, api::PaymentsRequest>, PaymentData<F>)>
+    where
+        F: 'b + Send,
+    {
+        let payment_method = payment_data.payment_attempt.payment_method;
```

Comment:
There should not be any uupdates happening in this operation, only post update tracker should be there right?